Running in AppVeyor produces invalid semver for NuGet

May be related to issue #635, but when building 'master' only GitVersion accepts the build # from AppVeyor which ends up producing an invalid semver according to NuGet on the build boxes.

```
74  INFO [11/25/15 21:42:01:99] Applicable build agent found: 'AppVeyor'.
 75  Executing GenerateSetVersionMessage for 'AppVeyor'.
 76  Set AppVeyor build number to '3.0.0-pre.12-(Build-30) (Build 31)'.
 77  Executing GenerateBuildLogOutput for 'AppVeyor'.
 78  Adding Environment Variable. name='GitVersion_Major' value='3']
 79  Adding Environment Variable. name='GitVersion_Minor' value='0']
 80  Adding Environment Variable. name='GitVersion_Patch' value='0']
 81  Adding Environment Variable. name='GitVersion_PreReleaseTag' value='pre.12-(Build-30)']
 82  Adding Environment Variable. name='GitVersion_PreReleaseTagWithDash' value='-pre.12-(Build-30)']
 83  Adding Environment Variable. name='GitVersion_BuildMetaData' value='']
 84  Adding Environment Variable. name='GitVersion_BuildMetaDataPadded' value='']
 85  Adding Environment Variable. name='GitVersion_FullBuildMetaData' value='Branch.master.Sha.e3a0fab9ea6fefd813041b1d3fcf037207645137']
 86  Adding Environment Variable. name='GitVersion_MajorMinorPatch' value='3.0.0']
 87  Adding Environment Variable. name='GitVersion_SemVer' value='3.0.0-pre.12-(Build-30)']
 88  Adding Environment Variable. name='GitVersion_LegacySemVer' value='3.0.0-pre12-(Build-30)']
 89  Adding Environment Variable. name='GitVersion_LegacySemVerPadded' value='3.0.0-pre12-(Build-30)']
 90  Adding Environment Variable. name='GitVersion_AssemblySemVer' value='3.0.0.0']
 91  Adding Environment Variable. name='GitVersion_FullSemVer' value='3.0.0-pre.12-(Build-30)']
 92  Adding Environment Variable. name='GitVersion_InformationalVersion' value='3.0.0-pre12-(build-30)']
 93  Adding Environment Variable. name='GitVersion_BranchName' value='master']
 94  Adding Environment Variable. name='GitVersion_Sha' value='e3a0fab9ea6fefd813041b1d3fcf037207645137']
 95  Adding Environment Variable. name='GitVersion_NuGetVersionV2' value='3.0.0-pre12-(build-30)']
 96  Adding Environment Variable. name='GitVersion_NuGetVersion' value='3.0.0-pre12-(build-30)']
 97  Adding Environment Variable. name='GitVersion_CommitDate' value='2015-11-25']
 98  INFO [11/25/15 21:42:03:34] CacheKey: C:\projects\psmsi\.git:refs/heads/master:e3a0fab9ea6fefd813041b1d3fcf037207645137
 99  INFO [11/25/15 21:42:03:34] Version not in cache. Calculating version.
 100  INFO [11/25/15 21:42:03:34] Applicable build agent found: 'AppVeyor'.
...
134AfterBuild:
 135  "C:\projects\psmsi\.nuget\NuGet.exe" pack "C:\projects\psmsi\src\PowerShell\PowerShell.nuspec" -Properties "SolutionDir=C:\projects\psmsi\;Configuration=Release;Platform=AnyCPU;Version=3.0.0-pre12-(build-30)" -OutputDirectory "bin\Release" -Symbols -NonInteractive -NoPackageAnalysis
 136  Attempting to build package from 'PowerShell.nuspec'.
 137  '3.0.0-pre12-(build-30)' is not a valid version string.
 138  Parameter name: version
 139C:\projects\psmsi\src\PowerShell\PowerShell.csproj(186,5): error MSB3073: The command ""C:\projects\psmsi\.nuget\NuGet.exe" pack "C:\projects\psmsi\src\PowerShell\PowerShell.nuspec" -Properties "SolutionDir=C:\projects\psmsi\;Configuration=Release;Platform=AnyCPU;Version=3.0.0-pre12-(build-30)" -OutputDirectory "bin\Release" -Symbols -NonInteractive -NoPackageAnalysis" exited with code 1.
 140Done Building Project "C:\projects\psmsi\src\PowerShell\PowerShell.csproj" (default targets) -- FAILED.
 141The target "_ConvertPdbFiles" listed in a BeforeTargets attribute at "C:\Program 
```
This didn't repro on other branches when pushing changes. I have no branch-specific info my _appveyor.yml_ other than to publish artifacts to nugget.org and powershellgallery.com, and `gitversion /showconfig` has the same ContinuousDeployment as every other branch / default.

I've looked through the source to see if there was a way to disable variations when run `%APPVEYOR%` is defined, but didn't find anything.

--- a/src/GitVersionCore/BuildServers/AppVeyor.cs
+++ b/src/GitVersionCore/BuildServers/AppVeyor.cs
@@ -13,14 +13,12 @@
 
         public override string GenerateSetVersionMessage(VersionVariables variables)
         {
-            var buildNumber = Environment.GetEnvironmentVariable("APPVEYOR_BUILD_NUMBER");
-
             var restBase = Environment.GetEnvironmentVariable("APPVEYOR_API_URL");
 
             var request = (HttpWebRequest)WebRequest.Create(restBase + "api/build");
             request.Method = "PUT";
 
-            var data = string.Format("{{ \"version\": \"{0} (Build {1})\" }}", variables.FullSemVer, buildNumber);
+            var data = string.Format("{{ \"version\": \"{0}\" }}", variables.FullSemVer);
             var bytes = Encoding.UTF8.GetBytes(data);
             request.ContentLength = bytes.Length;
             request.ContentType = "application/json";
@@ -39,7 +37,7 @@
                 }
             }
 
-            return string.Format("Set AppVeyor build number to '{0} (Build {1})'.", variables.FullSemVer, buildNumber);
+            return string.Format("Set AppVeyor build number to '{0}'.", variables.FullSemVer);
         }
 
         public override string[] GenerateSetParameterMessage(string name, string value)

